nalej/web
# Cluster lifecycle web ui

#### What does this PR do?
Implements the[ mockup design for the cluster lifecycle management](https://xd.adobe.com/spec/16138a00-1a69-42ff-4ba4-1c39bc50866b-6022/)

The current scope of this task includes:
- Modal windows with states. (Applications and resources view)
- A new button in the graph view to show the modal. (Applications and resources view)
- Contextual action buttons → Static contextual menu with those options: Edit, cordon, uncordon, more info(got to the existing view for now) and drain. We should show an error notification when it’s necessary. (Resources view)
- The removal of instances column and the new column with status color. (Resources view)
- Add the new colors to the graph
- Creation of tools service

#### Where should the reviewer start?
- tools.component.ts
- tools.service.ts
- applications and resources components
- backend.service.ts

#### What is missing?
The typo in the cluster status enum should change when task [NP-2448](NP-2448) will be done

#### How should this be manually tested?
1. Run the app
2. Go to Resources view
3. Verify the performance of the contextual menu in list card
4. Verify the performance of the graph and search box in the graph
5. Click on the right bottom button in the graph to see the status legend
6. Go to Applications view
7. Verify that everything runs ok and all the states are shown

#### Any background context you want to provide?

#### What are the relevant tickets?

- [NP-1929](https://nalej.atlassian.net/browse/NP-1929)

#### Screenshots (if appropriate)
![Captura de pantalla 2019-12-02 a las 16 43 57](https://user-images.githubusercontent.com/33664787/69973001-f8796c80-1522-11ea-9b5b-2f489de8e555.png)
